gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManagementTaskStreamImpl.java

```diff
@@ -135,6 +153,46 @@ public DagTask next() {
       }
   }
 
+  private void createStartDeadlineTrigger(DagActionStore.DagAction dagAction)
+      throws SchedulerException {
+    Pair<Optional<Dag.DagNode<JobExecutionPlan>>, Optional<JobStatus>> dagNodeToCreateTriggerFor =
+        this.dagManagementStateStore.getDagNodeWithJobStatus(dagAction.getDagNodeId());
+
+    TimeUnit jobStartTimeUnit = TimeUnit.valueOf(ConfigUtils.getString(this.config, DagManager.JOB_START_SLA_UNITS,
+        ConfigurationKeys.FALLBACK_GOBBLIN_JOB_START_SLA_TIME_UNIT));
+    long defaultJobStartSlaTimeMillis = jobStartTimeUnit.toMillis(ConfigUtils.getLong(this.config,
+        DagManager.JOB_START_SLA_TIME, ConfigurationKeys.FALLBACK_GOBBLIN_JOB_START_SLA_TIME));
+
+    long timeOutForJobStart = DagManagerUtils.getJobStartSla(dagNodeToCreateTriggerFor.getLeft().get(), defaultJobStartSlaTimeMillis);
+    long jobOrchestratedTime = dagNodeToCreateTriggerFor.getRight().get().getOrchestratedTime();
+    long reminderDuration = jobOrchestratedTime + timeOutForJobStart - System.currentTimeMillis();
```

Comment:
when this fires is that when it should have completed by? let's think about when we should calculate this time and who should store it? dagAction and/or task and/or reminder

Reply:
it feels like dagTask and proc should have an epoch that will contain this information. we can save it in the reminder jobDataMap if you overload the method to schedule the reminder